src/layout/rect.rs

```diff
@@ -166,6 +180,29 @@ impl Rect {
         }
     }
 
+    /// Resizes the `Rect` without modifying its position.
+    ///
+    /// Resizes the `Rect` according to the given offset without modifying its [`x`](Rect::x)
+    /// or [`y`](Rect::y) coordinates.
+    /// - Positive `width` increases the extent of the `Rect` along the x-axis, negative `width`
+    ///   decreases it.
+    /// - Positive `height` increases the extent of the `Rect` along the y-axis, negative `height`
+    ///   decreases it.
+    ///
+    /// See [`ResizeOffset`] for details.
+    #[must_use = "method returns the modified value"]
+    pub fn resize(self, resize_offset: ResizeOffset) -> Self {
```

Comment:
It's possible this make sense named "resized" rather than "resize" as the latter implies an action (e.g. a setter), where-as the former implies a non-modifying method that returns a new value. I'm not 100% sure on this, so it's worth looking at - what do we do that's consistent and what do other libraries that do this do for similar behavior. (To be clear, this is not a request to make this change, but more a request to think about it and push back if I'm wrong. I think this would be fine as `resize` if it's consistent with other things and we never want to have `Rect::resize(&mut self, resize_offset) {}`)

Reply:
I don't know but I also had the new idea to name it
`Rect::shift` and `layout::PositionDelta`, `Rect::resize` and `layout::SizeDelta`:

```Rust
let area = Rect::new(0, 0, 20, 30)
    .shift(PositionDelta { x: 0, y: 4 })
    .resize(SizeDelta {
        width: -5,
        height: 5,
    });
```

delta has the advantage that it just means difference while I associate offset more with a position than a size, so we could just keep `Offset` (but still rename `offset` as function name because it is imprecise):

```Rust
let area = Rect::new(0, 0, 20, 30)
    .shift(Offset { x: 0, y: 4 })
    .resize(SizeDelta {
        width: -5,
        height: 5,
    });
```

I am not sure what I prefer, the second option because it is concise or the first because it relates both types to each other.

Reply:
I also think so. With 0.29 just out there is some time anyways until the next tagged release.